mypy/semanal_typeddict.py

```diff
@@ -154,26 +156,24 @@ def analyze_typeddict_classdef(self, defn: ClassDef) -> tuple[bool, TypeInfo | N
             else:
                 self.fail("All bases of a new TypedDict must be TypedDict types", defn)
 
-        keys: list[str] = []
-        types = []
+        field_types = {}
         required_keys = set()
         readonly_keys = set()
```

Comment:
```suggestion
        required_keys: set[str] = set()
        readonly_keys: set[str] = set()
```
Not part of the PR but these could use explicit annotations as well.

Reply:
Any reason not to do this now as well? You've marked the comment as resolved without a resolution.